src/middlewares/validate.js

```diff
@@ -0,0 +1,27 @@
+const { validate, Symbols } = require('helpers/validator');
+const _ = require('lodash');
+
+const validateMiddleware = (validators = []) => async (ctx, next) => {
+  const payload = {
+    ...ctx.request.body,
+    ...ctx.query,
+    ...ctx.params,
+    [Symbols.PERSISTENT]: _.pick(ctx, ['request.body', 'query', 'params', 'state']),
```

Comment:
Do we need `state` here?

Reply:
We need state here
https://github.com/paralect/koa-api-starter/blob/master/src/resources/user/validators/update.validator.js#L36
I pass persistent data as the second param in validator function.
I think it is better to apply `Object.freeze` or `_.cloneDeep` to ctx and set the result to `[Symbols.PERSISTENT]` property